node-repository/src/main/java/com/yahoo/vespa/hosted/provision/applications/Applications.java

```diff
@@ -63,19 +63,24 @@ public void remove(ApplicationTransaction transaction) {
         db.deleteApplication(transaction);
     }
 
+    public record Lock(Mutex mutex, ApplicationId application) implements Mutex {
```

Comment:
FYI: There's already a type for this: `com.yahoo.config.provision.ProvisionLock`

Reply:
Thanks, I failed to find such a class even though I vaguely remembered there was one.  Fixed in https://github.com/vespa-engine/vespa/pull/29070